lib/thread/freertos/export/ecl/thread/semaphore.hpp

```diff
@@ -6,28 +6,32 @@
 
 #include <ecl/err.hpp>
 
+#include <atomic>
+
 namespace ecl
 {
 
-//!
 //! \brief FreeRTOS-based semaphore.
-//! \TODO: notes about FreeRTOS semaphore limitation
+//! This semaphore is specific. It can only be raisen or cleared.
+//! Default state is cleared, so calling wait() would block until someone
+//! will call signal()
 //!
-class semaphore
+class binary_semaphore
```

Comment:
Mutex is something that must be locked and unlocked within the same thread, effectively creating critical section. The Core already has one: 
https://github.com/forGGe/theCore/blob/develop/lib/thread/freertos/mutex.cpp#L5
Ref:
https://github.com/cjlano/freertos/blob/master/FreeRTOS/Source/include/semphr.h#L722